ecs-cli/modules/cli/compose/adapter/convert.go

```diff
@@ -43,6 +43,75 @@ const (
 	volumeFromContainerKey    = "container"
 )
 
+// ConvertToDevices transforms a slice of device strings into a slice of ECS Device structs
+func ConvertToDevices(cfgDevices []string) ([]*ecs.Device, error) {
+	devices := []*ecs.Device{}
+	for _, devString := range cfgDevices {
+		var device ecs.Device
+
+		parts := strings.Split(devString, ":")
+		numOfParts := len(parts)
+
+		switch numOfParts {
+		case 1:
+			device = ecs.Device{
+				HostPath: aws.String(parts[0]),
+			}
+		case 2:
+			device = ecs.Device{
+				HostPath:      aws.String(parts[0]),
+				ContainerPath: aws.String(parts[1]),
+			}
+		case 3:
+			permissions, err := getDevicePermissions(parts[2])
+			if err != nil {
+				return nil, err
+			}
+			device = ecs.Device{
+				HostPath:      aws.String(parts[0]),
+				ContainerPath: aws.String(parts[1]),
+				Permissions:   aws.StringSlice(permissions),
+			}
+		default:
+			return nil, fmt.Errorf(
+				"Invalid number of arguments in device %s", devString)
+		}
+
```

Comment:
What do you think about DRYing up this code a bit? Since `device` is instantiated in line 50 and some of the fields are repeated, maybe something like:

```
		hostPath := parts[0]
		device.SetHostPath(hostPath)

		if numOfParts > 1 {
                        containerPath := parts[1]
			device.SetContainerPath(containerPath)
		}

		if numOfParts > 2 {
			permissions, err := getDevicePermissions(parts[2])
			if err != nil {
				return nil, err
			}
			device.SetPermissions(aws.StringSlice(permissions))
		}
		if numOfParts > 3 {
			return nil, fmt.Errorf(
				"Invalid number of arguments in device %s", devString)
		}
```